File: cpp/easy/0242_valid_anagram.cpp
// https://leetcode.com/problems/valid-anagram/submissions/1258401000/
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }
        unordered_map<char, int> letters;
        for(auto letter : s){
            letters[letter]++;
        }
        for(auto letter : t){
            letters[letter]--;
        }
        for(auto &entry : letters){
            if(entry.second != 0){
                return false;
            }
        }
        return true;
    }
};

File: python/easy/0242_valid_anagram.py
# https://leetcode.com/problems/valid-anagram/description/
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = {}
        if len(s) != len(t):
            return False
        for i in range(len(s)):
            if s[i] in count:
                count[s[i]] += 1
            else:
                count[s[i]] = 1
            if t[i] in count:
                count[t[i]] -= 1
            else:
                count[t[i]] = -1
        
        for key, val in count.items():
            if val != 0:
                return False
        return True
